src/common/Throttle.cc

```diff
@@ -725,31 +723,99 @@ TokenBucketThrottle::~TokenBucketThrottle()
 
 void TokenBucketThrottle::set_max(uint64_t m) {
   std::lock_guard<Mutex> lock(m_lock);
-  m_throttle.set_max(m);
+  if (0 == m && 0 != m_avg) {
+    m_throttle.set_max(m_avg);
+  } else {
+    m_throttle.set_max(m);
+  }
 }
 
 void TokenBucketThrottle::set_average(uint64_t avg) {
-  m_avg = avg;
+  {
+    std::lock_guard<Mutex> lock(m_lock);
+    m_avg = avg;
+
+    if (0 == avg) {
+      // this means No Limit
+      m_tick = 1000;
```

Comment:
so we set the tick to 1000ms, you can add more comment here to explain what does "1000" means.

Reply:
All right.